packages/react-magma-dom/src/components/TreeView/TreeItem.tsx

```diff
@@ -242,6 +256,107 @@ export const TreeItem = React.forwardRef<HTMLLIElement, TreeItemProps>(
           : checkedStatus === IndeterminateCheckboxStatus.checked
         : null;
 
+    const [isInsideTreeItem, setIsInsideTreeItem] = React.useState(false);
+    const treeItemRef = React.useRef<HTMLLIElement>(null);
+    const focusTrapElement = useFocusLock(isInsideTreeItem);
+    const interactiveElements =
+      'button, [role="button"], input, select, textarea, a[href], [tabindex]:not([tabindex="-1"])';
+
+    const getInteractiveElements = (
+      container: HTMLElement,
+      selector: string
+    ): HTMLElement[] => {
+      return Array.from(
+        container.querySelectorAll<HTMLElement>(selector)
+      ).filter(
+        el =>
+          !el.hasAttribute('tabindex') ||
+          (el.tabIndex !== undefined && el.tabIndex >= 0)
+      );
+    };
+
+    const handleLabelAndAdditionalContentKeyDown = (
```

Comment:
Can we add a comment explaining what this is all doing? I understand it today but we'll want to remember in the future why we built it this way